src/braze-upload/lambda/lambda.ts

```diff
@@ -1,16 +1,57 @@
+import {Pool, QueryResult} from "pg";
+import SSM = require("aws-sdk/clients/ssm");
+import {createDatabaseConnectionPool, fetchReferralData} from "../lib/db";
+import {getParamsFromSSM} from "../lib/ssm";
+
 const AWS = require('aws-sdk');
 const acquisition_types = require('../gen-nodejs/acquisition_types');
 const serializer = require('thrift-serializer');
 
-export async function handler(event: any, context: any): Promise<null> {
-    console.log("events:", JSON.stringify(event));
-    event.Records.map(record => {
-        serializer.read(acquisition_types.Acquisition, record.kinesis.data, function (err, msg) {
+const ssm: SSM = new AWS.SSM({region: 'eu-west-1'});
+const dbConnectionPool: Promise<Pool> =  getParamsFromSSM(ssm).then(createDatabaseConnectionPool);
+
+interface Event {
+    Records: {
+        kinesis: {
+            data: any
+        }
+    }[]
+}
+
+const getReferralCodeFromThriftBytes = (rawThriftData: any): Promise<string | null> =>
+    new Promise((resolve, reject) => {
+        serializer.read(acquisition_types.Acquisition, rawThriftData, function (err, msg) {
             if (err) {
-                console.log("error", err)
+                reject(err);
+            }
+            console.log("event:", JSON.stringify(msg));
+
+            const referralCodeParam = msg.queryParameters.find(qp => qp.name === 'referralCode');
```

Comment:
Could `referralCodeParam` be given its own class, with an `isValid()` method called here, such that it can then be passed in to `fetchReferralData` more safely?

Reply:
what type of validation are you suggesting?

Reply:
min length; the two formats: Braze nanoseconds as a string, and support-fe base36-encoded milliseconds...

Reply:
See https://github.com/guardian/contributions-referrals/pull/9